bumble/device.py

```diff
@@ -2808,11 +2808,23 @@ async def get_pin_code():
                 )
 
                 if pin_code is not None:
-                    pin_code = bytes(str(pin_code).zfill(6))
+                    # TODO: make pin code input more flexible,
+                    # so far only numbers are allowed
+                    pin_code = bytes(str(pin_code), encoding='utf8')
+                    pin_code_len = len(pin_code)
+                    if pin_code_len >= 0x10:
+                        logger.warning(
+                            'pin_code is longer than 16 bytes, truncated to 16'
+                        )
+                        pin_code_len = 0x10
+                        pin_code = pin_code[:16]
+                    else:
+                        pin_code = pin_code + b'0' * (16 - pin_code_len)
```

Comment:
Shouldn't it need to be a prefix padding ?
`pin_code = bytes(str(pin_code).zfill(16))` should do the trick.
I think 6 bytes is enough

Reply:
The way I read the spec, the PIN code is a string, up to 16 bytes. So 0-padding past PIN_Code_Length should be right.

Reply:
Whether padding before the pin code or after the pin code is something I am not sure.

Reply:
To summarize, we should pad b'0x00's, but as padding is done for us, in final version of this change, we do not pad.